Exo Station - Upstream Playtest Version

SlamBamActionman opened the pull request:
<!-- Guidelines: https://docs.spacestation14.io/en/getting-started/pr-guideline -->

## About the PR
<!-- What did you change? -->

This PR introduces a new station map to the game: **Exo** (previously Xeno), a map themed around NT having taken over an alien vessel and turning it into a proper station.

![Exo-0](https://github.com/user-attachments/assets/3f6b44ae-3206-4a35-8908-a9672f2d429e)

It features a lot of departmental quirks, custom Cargo shuttle/Evac shuttle, and unique tiles & walls.
<b>This PR does not add the map into rotation and is intended for playtest purposes.</b>

**Massively huge thanks to JuneSzalkowska, obscenelytinyshark and Kezu for their help spriting custom assets for the station!**

## Station information

**Mapname:** Exo - Exomorph Vessel "Conquest" EXO-##-###
**Playercount:** 25-60 (Upper limit as of yet untested)

### Cargo Shuttle
![image](https://github.com/user-attachments/assets/41b37102-b334-4a51-8e14-c265060ffd39)

### Evac Shuttle
![image](https://github.com/user-attachments/assets/88394767-f73a-4aad-bb34-a9583cdf08c4)

## Specific area descriptions

<details>
<summary><b>In-depth summary of areas (click to open)</b></summary>
<br>

This section will go through summaries of different departments and design decisions made; the map features a lot of quirks in how the departments are laid out to promote different gameplay areas of the game, which is why this section will go through various choices made and why.

<b>Note that any images here may not be kept up to date if the PR gets more commits; the main station image will always be updated however.</b>

## Engineering, AME and Singularity Confinement

Engineering is in its layout fairly standard; located in the southwestern point of the map, it's front is in a medium-traffic area with its sides being less-so, flanked by dorms and library maints respectively. It is smaller in overall size due to the fact Atmospherics has been split off to a separate section of the station, leaving Engineering to be exclusively for Engineers and CE. The area is mostly set up by NT, hence its use of normal walls & tiles.

![image](https://github.com/user-attachments/assets/ba9da2c0-5a08-43e3-abc8-07cceafb8e73)

One of the most immediate changes for Engineering is the existence of the *public AME*. The AME space is situated in a large high-traffic area in the central crossing of Chapel, Vault, Science and Medical. This may make it appear as a very vulnerable place to have the AME, however due to the high traffic it becomes very easy for crew to spot if the AME is getting overloaded or if someone is in the process of stealing the AME parts. To assist with this there has been materials provided to create a barrier around the AME (reinforced directional windows have become a playtest standard).

![image](https://github.com/user-attachments/assets/71d04a79-d809-4ea9-9b23-531d9e7cc9b3)

Due to how exposed the AME is, Engineers may still be uncomfortable running it. To offset the AME's downsides, the station has a very efficient and prepared setup for its Singularity/Tesla containment. It's set up 50% for Singularity, 50% for Tesla, with colored markings to assist newer engineers in setting up a specific containment. The meteor protection is not fully reinforced walls, however due to its multiple layers it should require several hits for a meteor to break through.

![image](https://github.com/user-attachments/assets/35ec807c-34d9-4185-992d-bb00a83b5b1a)

## Atmospherics

Atmospherics has been separated from the Engineering department into an in-station setup with a moderately sized TEG area. There is only one roundstart burn chamber, so it may become a choice whether a TEG or Frezon/Tritium production is necessary. It's position in the western part of the station should make engineering players able to respond to spacings to multiple locations on the station fairly quickly. Atmospherics also has the Tech Vault in its area, which requires Engineering + Command access to open.

![image](https://github.com/user-attachments/assets/8cb6d39e-ef27-466b-bd7e-d4566a533b32)

## Medical & Viro / Morgue

Medical is located in the northern-center part of the station, with its main entrance in the southwestern corner. It's set up by NT, which is why it features regular reinforced walls, windows and tiles. The main treatment bay is fairly small, though cleaned up to provide plenty of space for medicine and chem storage. More emphasis is given to the surgery suite in the back of medbay in preparation for newmed. Of note is the Chemistry chute in the east section which gives access to Chemistry's backroom; this will be expanded upon in the Chemistry section.

There's two back areas for doctors to retreat to if necessary (generic storage & changing room/med storage). CMO has their office in the northern section, sharing a door to EVA should retreat be necessary (while also providing a backdoor for antags to utilize).

![image](https://github.com/user-attachments/assets/468904e5-4a13-4685-a172-a0027c347deb)

Virology is separated by a (possible-to-close) corridor east of Medical. This is also where the morgue is located. Virology provides a shortcut to the northeast side to the station, which otherwise requires a bit of a walkaround. It's a fairly visible area from the outside save for the patient room, making it not as optimal for antag prep as virology tends to be. While Virology shares its wall with the visitor dock the corridor is fairly low-traffic however.

The morgue is fairly standard, though it has a conveyor belt that goes around the station, passing by Security and ending up in the Kitchen's freezer. 

![image](https://github.com/user-attachments/assets/2608342a-b5da-4925-84df-1edabe4ce8c3)

## Chemistry & Botany

One of the most controversial changes is separating Chemistry from Medical. Chemistry has been moved to the south-center of the station, being very close to Cargo and Atmospherics, a fairly high-traffic area. This means Chemistry has to rely more on communication with Medical to coordinate medicine. While the separation in physical distance is far, it is not practically distant due to the addition of separate disposals bin systems in the Chemistry backroom. Chemists are able to send chem jugs either directly through the bins separately, in a big or jump in themselves (if there is a bin on both ends, containers do not spill). This disposals system is intentionally positioned underneath walls and doorways, making it harder to sabotage. Of note is that both disposal bins in Chemistry lead to the backroom, which has been intentionally separated from the main work area in case a Botanist/Doctor uses the system to travel to Chemistry.

Chemistry also has a small mini-medbay; this was added after concerns that Chemistry would be too insular. With its position near Cargo, Chemists become the first point of contact for injured Salvagers and are able to provide quick medical help if there is a low medical playercount.

![image](https://github.com/user-attachments/assets/ed07ad3b-2b29-4a48-8c31-23a78b2df773)

Botany is located fairly close to Chemistry, northwest of it opposite to HoP's office. It has a fairly standard setup, though of note is the Chemistry disposals bin (Botany-locked, as the botany side gives access to the entirety of Botany, while the Chemistry side only gives access to the backroom) and an Atmos setup for that rare Frezon mutation. Botany also gets responsibility for the animals.

![image](https://github.com/user-attachments/assets/d895abe5-6f62-4944-8489-4785a1f35d45)

## Science & Chapel

Science is fairly standard in its main area (foreshadowing). It's located in the northwestern part of the station and is the closest department to both Evac and Arrivals, with its front desk on the northeastern section. The front desk has an area for setting up additional machines and access to the basic Science equipment. There's a visible surgery and robotics area that have access to the AME room. The server is in a closed-off area that requires a Ninja to go through the main Science area to reach, however it is also vulnerable to having its (reinforced) back wall deconstructed.

![image](https://github.com/user-attachments/assets/590cafb5-b7be-48e2-94af-c5ffc5b4b54c)

The main quirk of Science is its compromise in sharing space with the Chapel; the Chaplain set up shop to worship the ancient alien artifacts, but then Science had to try to take over in the name of "research". The choice to have this shared space is to encourage interaction and roleplay between the Chaplain and Science, two roles that rarely have much reason to interact otherwise. 

The artifact chamber is surrounded by multiple layers of reinforced plasma glass and walls, but also has an additional set of doors positioned on the southern side with Chapel access. The chapel itself puts the artifact chamber as its main attraction, with the Chaplain's room being in the southern side. The Chapel is also a direct route to the Bridge entrance from the eastern side of the station, making it a medium-traffic area. (Not included in the render is the chapel altar, which is located in the chamber). 

![image](https://github.com/user-attachments/assets/64346bae-67a1-4c05-ad89-7e9ca5e2efd9)

## Security

Security is located in the far east section of the station. It comes with 3 cells and acts as a blocker for movement in the eastern section of the station for non-Security personnel. 

One of its key design features is the choice to split up the Armory in *two*. The mini-Armory is easily accessible in the main area of Security, has a few guns and a suit, and contains the SecFab. The main Armory is located on the outside of Security. While this may make it seem like it is more susceptible to illegal entrance, it's protected with a blast door, 3 Security-locked doors and locked lockers, and is positioned in a medium-traffic area as its set by Security's entrance and in view from Warden's office/Security's breakroom. This design should make Security less weak towards Nukie armory raid since there are two areas that need to be hit, and when the main Armory has to open there is likely a threat large enough that Security will be able to monitor crew access (or it's dire enough that crew should arm anyways). 

![image](https://github.com/user-attachments/assets/59de2f08-bb35-4e9e-9ca7-f018d125beaf)

The brig contains three cells. It's retrofitted from a meat processing area; running from the Morgue to Kitchen is a conveyor belt for bodies, with connections to Security via the cells. This connection is defended by loud hostile xeno turrets and custom airtight flaps that take longer than normal to unanchor, giving Security time to respond to any break-out attempt via sound and visibility. 

Cells have some basic food, though drink is provided via saline syringes. Of note is the NO2 canister located in the northern section (which also has an access-locked disposals bin). Security has the ability to sedate prisoners using this, with the system set up to correct values by default.

![image](https://github.com/user-attachments/assets/10d1c135-bdcc-4cc7-b055-19abf8f756e8)

## Perma

Perma has one main unique feature; by default it is not separated from maintenance by any means, with an open entrance on its southern side. It becomes Security's responsibility to activate the Containment Field Generators, causing a containment ifeld to form, blocking off this opening. Touching the containment field causes you to bounce back and take minor blunt damage, and is a surprisingly effective wall that can't be deconstructed or bypassed (to my knowledge; testing has not shown any weaknesses in the wall itself yet). Getting rid of the emitter becomes a possibility, but that becomes about as difficult as any prison break. 

The emitter has been positioned such that you can exist in perma without hearing the constant sound of it firing, as this was a critique to its initial design.

![image](https://github.com/user-attachments/assets/d4e6c48c-bc39-47b3-a188-82633198ee59)

## Cargo & Salvage

Cargo is an addition to the ship sponsored by DonkCo, your favorite microwave food brand! For normal crew there is a very small reception area separating them from Cargo via glass and shutters; instead of Cargo Techs dragging out crates and items manually, there is instead a conveyor belt system used to transport items to crew. The traversal for Cargo Techs into/out of the station comes via two paths on either side of Cargo, with the west path being the main one. The southeast has Salvage suit storage and an airlock out to the salvage satellite.

Cargo also has a custom Donk-branded Cargo Shuttle, with a large airlock opening for transporting large volumes/sizes of goods.

![image](https://github.com/user-attachments/assets/72ac9ac7-9efd-4e37-bdff-af99ffafdf64)

Salvage is separated from the main part of cargo into their own little satellite. It contains the basic necessities and should provide ample access and work area.

![image](https://github.com/user-attachments/assets/46200e00-39ad-410e-8300-6356d9b9dfb8)

## Bar & Kitchen

The bar has a militaristic aesthetic merged with a roadside diner serving layout; with the Kitchen separated from the main eating area by the bar, the Chefs and Bartenders are made to interact via the Bartenders serving the food for the Chefs. The Freezer has some roundstart xeno meat to fit the map, and also acts as the endpoint of the Morgue conveyor.

![image](https://github.com/user-attachments/assets/6d63a2d5-c31a-4801-94f5-b0ef1ed44d65)

## Small Fun Things!

### Library

The back of the library acts as a shortcut between Cargo and Engineering front/Command, which should provide some throughfare.

![image](https://github.com/user-attachments/assets/dc1d257b-c85c-43c1-aeaa-2d898e01dffe)

### The Lone Praetorian

In the easternmost maintenance, east of Security, there is a barricaded completely dark maze-like area. In this area a special mob, the Lone Praetorian, spawns. It is a fairly non-lethal mob and instead acts as a jumpscare for any unaware crew entering that section of maintenance. There is not much loot in there (some few maintenance locker and a salvage loot crate) to discourage experienced players beelining to it roundstart. The Praetorian deals fairly low damage and injects the target with stimulants/tranexamic acid; if it succeeds in a jumpscare it can disorient players, so this design helps retreating and protects against a Tider dying in maintenance without coords on. 

![image](https://github.com/user-attachments/assets/e7973a42-2721-48e9-b261-cf7c700536a8)

### Gun Bays

The station has two old gun bays, one which is public access and the other being where the Clown and Mime have set up their rooms. Their areas act as access to the western solars, but the gun bays themselves are set up with still functioning ship guns. *These ship guns have been modified for this map to be a lot less effective, shooting and recharging slower, and being much heavier and thus harder to move.* For the Clown/Mime room, the firing button is in the Mime room. The Clown thankfully spawns behind some layers of plasma glass for protection. Each gun bay also has a button locked behind Externals access that opens the gun bay doors.

![image](https://github.com/user-attachments/assets/20272338-a44a-46a0-a1e7-86aa72336673)
![image](https://github.com/user-attachments/assets/4e887a05-4d89-4e7d-b4d7-9c9ec49b3dd3)

### Arrivals

Arrivals is the attack ship NT used to breach and enter the station during the initial assault. It is visually distinct from the rest of the station and contains some basic supplies, as well as the cryo beds. This separation should make it less focus for attacks. The design and layout of Arrivals is very similar to the Evac shuttle.

![image](https://github.com/user-attachments/assets/9d2ec8c9-0b93-44b9-b1fe-f47bf00ba5da)

### Maints bar

The maints bar is set up with some spare materials in the backroom, making it a quick and easy process to set it up with your own choice of how it should be designed! Easy customization!

![image](https://github.com/user-attachments/assets/2f7b9c1b-d160-47c4-97c1-4c362fe086bc)

### Vault & Nuke

The nuke is located in the center of the station, and is visibly out in the open. Nukies either have to breach via the vault on the west side, or burst through the resistant reinforced glass directly. Since the Nuke is in the middle, Gravity/Bridge in the west and Security/Armory in the east, Nukies should have difficulty hitting all three points, giving them a choice on how to approach.

![image](https://github.com/user-attachments/assets/dd52c64d-4519-4589-a141-50bb54a5299f)

</details>

## Technical details
<!-- Summary of code changes for easier review. -->

<details>
<summary><b>New prototypes (click to open)</b></summary>
<br>

**Xeno floor tile**
Decorative floor tile, has a squishy sound effect when walked over.
![xeno_flooring](https://github.com/user-attachments/assets/ddb70ef7-3eb2-4bfe-84e8-961921901f52)

**Dark tech maint tile**
A darker version of the tech maint tile, can be created via the tile cutter machine.
![tech_maint_dark](https://github.com/user-attachments/assets/67e92a66-ab66-4f84-9968-af005bda0fed)

**Solid chitin & Reinforced chitin**
Decorative walls, has the same properties as normal solid walls/reinforced walls. Has a design meant to support multi-thick walls. Deconstructs into steel, can not be constructed.
![full](https://github.com/user-attachments/assets/f9f3105d-2d28-4285-83ad-1a7ee07f32f5)
![rgeneric](https://github.com/user-attachments/assets/33d4a220-95fc-4ecc-9731-7f5e1dc384d3)

**Full Overlay decal**
A full square decal, used to color grass tiles in Hydroponics.
![full](https://github.com/user-attachments/assets/6f4acfb6-c466-4548-ba7d-9d68758586a6)

**Xeno Fire Alarm & Xeno Air Alarm**
Re-colored version of the basic fire & air alarms. Were necessary as the normal designs were unsuitable for the station environment.
![image](https://github.com/user-attachments/assets/d3822852-d8fe-429b-b8f9-3be45e53f0e1) ![image](https://github.com/user-attachments/assets/5e6f9b13-ce47-4581-bf21-434357a79e64)

**Rusted Thruster**
A decorative item to act as broken thrusters on the back of the ship; made into decorative as to not provide 10+ thrusters to the station roundstart.
![base](https://github.com/user-attachments/assets/d2c0434d-c680-4c56-93f0-ff8792b1cf0a)

**PuddleBlood, PuddleBloodSmall, PuddleFluorosulfuricAcid, PuddleFluorosulfuricAcidSmall**
New puddles with blood & fluorosulfuric acid (xeno blood) reagents

**SpawnMobXenoQueen & Lone Praetorian**
A mob spawner for a custom Praetorian mob. This Praetorian has higher-than-normal health but deals relatively minor damage and injects the target with minor stimulants & tranexamic acid; this makes it easy to run from if surprised and not lethal to fight, which is its intended function due to its spawn location.

**CartridgePistolSpent**
A prototype of a spent .35 cartridge, used as a decorative item.

**XenoStation Parallax**
A custom parallax for the station; basic, but tinted purple!
![image](https://github.com/user-attachments/assets/5e1cf385-3da0-4779-bbf2-1d1a7cd0fb00)

</details>

## Additional information

This map has gone through multiple playtests; it has been playtested on the following servers:
- Sector Umbra, an HRP server with 20~ players.
- Harmony, an MRP server with 50~ players.
- One-time server with players from Wizden, announced as an event via the official SS14 Discord, 30~ players.

This has included multiple rounds, with multiple playtests, and Harmony currently runs it as a map that can be ran at admin discretion. Reception has been very positive, with friction in the intended areas (e.g. public AME, Chemistry being separated) at appropriate and intended levels. Engineering's Singularity setup and Cargo's layout has been noted as very appreciated by players, and quirks of the map (Morgue conveyor, Perma containment, Xeno spawn) have shown to be enjoyable.

## Known bugs

- The perma containment field becomes invisible when standing at the far western side; this is due to the containment field entity is located at the initiating generator, and if that gets unloaded in PVS the entire field is hidden. It still functions as normal though.

## Requirements
<!-- Confirm the following by placing an X in the brackets [X]: -->
- [x] I have read and am following the [Pull Request and Changelog Guidelines](https://docs.spacestation14.com/en/general-development/codebase-info/pull-request-guidelines.html).
- [x] I have added media to this PR or it does not require an ingame showcase.
<!-- You should understand that not following the above may get your PR closed at maintainer’s discretion -->

## Breaking changes
<!-- List any breaking changes, including namespaces, public class/method/field changes, prototype renames; and provide instructions for fixing them.
This will be posted in #codebase-changes. -->

**Changelog**
<!-- Add a Changelog entry to make players aware of new features or changes that could affect gameplay.
Make sure to read the guidelines and take this Changelog template out of the comment block in order for it to show up.
Changelog must have a :cl: symbol, so the bot recognizes the changes and adds them to the game's changelog. -->

:cl:
- add: A new station, Exo, has been added to the map pool.

ArcaneOcto commented:
Ok but perma having a containment field has to be the coolesf thing ever.

Randomplayer2or3 commented:
Honestly might as well expand the morgue to kitchen belt to a full Goon-style belt hell that moves cargo to departments for how long the belt is.

TurboTrackerss14 commented:
![image](https://github.com/user-attachments/assets/768e927d-8ea1-4759-9d0f-d0022e0bddd5)
Laser box exists.

![image](https://github.com/user-attachments/assets/38fecd91-9146-4e36-94f8-a837c447a88b)
![image](https://github.com/user-attachments/assets/4a7384d2-c0f8-41a9-8b9c-3e197fb33179)
The DO NOT MAP behind you

![image](https://github.com/user-attachments/assets/c5c18e07-8269-4832-98d7-2efbb5eb12ce)
presented without comment

![image](https://github.com/user-attachments/assets/621e3af3-3ab8-4ee8-b808-082188aa4462)
QM flash and weld resist shiftstart.

![image](https://github.com/user-attachments/assets/b751094d-a8f5-4c55-a8e3-086bd9e46fa0)
This is stolen valor from barratry maint bear how dare you. 

![image](https://github.com/user-attachments/assets/2854260c-be59-489c-8f22-b7d856b375dd)
TEG cooling loop could do with being highlighted a little more, maybe decals. I missed this entire on my first sweep of the map render.

![image](https://github.com/user-attachments/assets/49962961-ddf4-4e8e-ae13-720f53676b7d)
Already a 50% spawner in the hop locker

![image](https://github.com/user-attachments/assets/e3b7412d-7967-48b9-953e-4b72fbe4c885)
If we are gunna go "tcomms in each department substation" we should really be consistent. (this is whatever cause its sec but command still has theirs in substation room) 

SpawnMobXenoQueen should be SpawnMobXenoPraetorian because its not a queen ™️

SlamBamActionman commented:
Thanks for reviewing! I'll address the points that need clarification, a lot of the other stuff are fixes I fully agree with. 👍 

> ![image](https://private-user-images.githubusercontent.com/130304754/382599587-38fecd91-9146-4e36-94f8-a837c447a88b.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzA2NjY2MDEsIm5iZiI6MTczMDY2NjMwMSwicGF0aCI6Ii8xMzAzMDQ3NTQvMzgyNTk5NTg3LTM4ZmVjZDkxLTkxNDYtNGUzNi05NGY4LWE4MzdjNDQ3YTg4Yi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQxMTAzJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MTEwM1QyMDM4MjFaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0wZDEzM2NkZDJhMTU0MTUzM2M1MGI4NTZkZmU4NzI4YjBkMjgxNjI2ZTA3YmQ3NWJiNmUyNmM5MzcyZTVjMDdiJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.Kh-9EpuN4UI8C4Fh7n2EECoATedZ-jrwXtJ_0l_5PH4) ![image](https://private-user-images.githubusercontent.com/130304754/382600782-4a7384d2-c0f8-41a9-8b9c-3e197fb33179.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzA2NjY2MDEsIm5iZiI6MTczMDY2NjMwMSwicGF0aCI6Ii8xMzAzMDQ3NTQvMzgyNjAwNzgyLTRhNzM4NGQyLWMwZjgtNDFhOS04YjljLTNlMTk3ZmIzMzE3OS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQxMTAzJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MTEwM1QyMDM4MjFaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1mYjY0ODJhODMyY2FhN2RiOTVlNGQ4ZmIyN2Y0MWZhNThjZTJmNzQ2Mzc4MzBjZDE3M2FjNzU1YWRkY2QyYjJmJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.uo4sA1Y7mQSxbSsUxQWwV4TFZ0HyIyVZ3QZbPHvRJbA) The DO NOT MAP behind you

I've written about this in the "Gun Bay" summary but to restate here; the dematerializers mapped to the station are modified for the map, featuring a custom name with slower firing speed, longer reload time and increased weight making them harder to relocate. The map initially had the default guns but as playtests showed that became abusable once more items were given structure damage in a recent PR. These scaled-back guns should be more appropriate as a map entity; I'll pull up the actual numbers soon.

> ![image](https://private-user-images.githubusercontent.com/130304754/382599605-c5c18e07-8269-4832-98d7-2efbb5eb12ce.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzA2NjY2MDEsIm5iZiI6MTczMDY2NjMwMSwicGF0aCI6Ii8xMzAzMDQ3NTQvMzgyNTk5NjA1LWM1YzE4ZTA3LTgyNjktNDgzMi05OGQ3LTJlZmJiNWViMTJjZS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQxMTAzJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MTEwM1QyMDM4MjFaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT05ZWMyYWViNzQxNWVkYmQyMTVhNzM2NzhjNGJiNzc1ZjQzYWQyYzExOTQxMWUzMmE3ZDVmZTJhZTFkYjhmMjY5JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.-i87dotTd0r_Al05SIvYho7PkObttXJKc0vgIRnQvwM) presented without comment Edit: never mind presenting with comment. Why are you adding a "cream" jug into this room specifically...

Totally agree it should be removed; this is a vestige poking fun at the fact the raunchy poster is still in the game and was intended to be replaced once the poster was, but since that discussion is still ongoing it slipped past.

> ![image](https://private-user-images.githubusercontent.com/130304754/382599700-621e3af3-3ab8-4ee8-b808-082188aa4462.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzA2NjY2MDEsIm5iZiI6MTczMDY2NjMwMSwicGF0aCI6Ii8xMzAzMDQ3NTQvMzgyNTk5NzAwLTYyMWUzYWYzLTNhYjgtNGVlOC1iODA4LTA4MjE4OGFhNDQ2Mi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQxMTAzJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MTEwM1QyMDM4MjFaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1iMGMzN2ZjZGYzNGRkM2QxMWQ0NmM0MmFkMDViMjY0MTEyM2Y0N2RjZTAzNTM3MjNhOWIzN2E0MTA4MjA1NDUxJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.DyWnpwv2BaK6P1Ou4h0-ZzLwCUIHrleFMFZUMUacDg4) QM flash and weld resist shiftstart.

Didn't know these glasses had that property! Thanks for pointing it out!

> ![image](https://private-user-images.githubusercontent.com/130304754/382600227-2854260c-be59-489c-8f22-b7d856b375dd.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzA2NjY2MDEsIm5iZiI6MTczMDY2NjMwMSwicGF0aCI6Ii8xMzAzMDQ3NTQvMzgyNjAwMjI3LTI4NTQyNjBjLWJlNTktNDg5Yy04ZjIyLWI3ZDg1NmIzNzVkZC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQxMTAzJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MTEwM1QyMDM4MjFaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1lMzU0ZWEyNjY3YzZlZmQxYWM4NmVkMjE0YzQ0Y2FlMzViZjQ5YTNkMWNhN2NkOWI0NzRjZDI3YmY2N2RhZDdmJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.r9kVRQj4I71pU1G8Qoiy4kGRYmtTa7Xn6tqsnCTecnU) TEG cooling loop could do with being highlighted a little more, maybe decals. I missed this entire on my first sweep of the map render.

I've tried to mark it a bit more (the coldsauce bottle was put there for a reason) but I'll look into ways to make it stand out more. Iirc there weren't many good decals indicating an atmos cold loop...

ArtisticRoomba commented:
I love the map design; I can really see where NT kinda just crammed parts of other stations into the poor ship. The contrast between xeno and NT hull is very, very cool in some spots where NT very obviously bolted departments onto the ship (cargo, for example). The plasma is a very good thematic touch (the xenos yearn for the plasma). The story being told through the map alone is amazing, jsut as an example I really like the """decontamination buffer""" cargo crates have to go through.

My few gripes with Engi/Atmos (as other people have said) are:
- Atmos is a bit cramped for a 70 pop map.
- Like Turbo said, the cold loop on the TEG could be pointed out better. Otherwise I see this suffering from the same problem as Cog, where people don't know there's a pre-ran space cooling loop.
- The public AME is poor wizden LRP self-antag bait but otherwise I find the change hilarious.
- The CE needs a mapped extra RCD and respective charges, similar with other maps for consistency.
- Engi needs a more obvious mats vault available to them.

Otherwise, amazing singulo/tesla containment design, I love it.

Possibly some more thematic changes to the map:
- Add xeno blood in and near maints. They're in the walls.
- Add grenade blast decals in maints rooms to show where the marines flushed the bugs out.
Additionally:
- More signs could be placed. Vault is missing one.

SlamBamActionman commented:
Thanks for the feedback! Environmental storytelling was a large part of this map, so I'm glad it's coming across.

> * Atmos is a bit cramped for a 70 pop map.

I might consider this more in the range of 25-60 tbh; I don't want to set a too-low cap since I just haven't had a chance to play it with that many (worth considering now that Wizden servers have a temporary max cap at 70 and the very big stations still seem to run fine). Atmos does have potential to expand fairly easily south and east, it's mostly a question of how cramped it actually ends up being. 

> * The CE needs a mapped extra RCD and respective charges, similar with other maps for consistency.

Didn't know this! Thanks for pointing it out!

> * Engi needs a more obvious mats vault available to them.

Materials are contained in the crates on west and east side of Engineering; it doesn't show well on the map renderer since they're just crates, but materials are there and should be locatable ingame.

> * Add xeno blood in and near maints. They're in the walls.
> * Add grenade blast decals in maints rooms to show where the marines flushed the bugs out.

I should! Right now the bloodbath is mostly constrained to the breaching area and the maints maze, but maybe some more stuff can be introduced in certain locations. It was just so much work individually rotating each bullet casing...

>   Additionally:
> * More signs could be placed. Vault is missing one.

Oops!

SlamBamActionman commented:
The map has been updated to Beta 1.3.5, with the following changes:

- Atmos has been slightly modified to (hopefully) make the cold loop at least somewhat clearer.
- The "Full" overlay decal has been removed and replaced with the "Square" markup, since there's no point in having two full square decals.
- QM got cheap sunglasses instead of mercenary glasses.
- Xenomaid room will annoy Security instead of Admins
- Ian backpack removed
- Captain's laser has been boxed
- CE got RCD ammo in their office (the RCD was already mapped, just hidden in the map render)
- Added signs in places!
- Probably some small stuff I forgot.

TurboTrackerss14 commented:
*coughs* Don't forget this 
>SpawnMobXenoQueen should be SpawnMobXenoPraetorian because its not a queen ™️

slarticodefast commented:
The map needs bombing targets mapped for the spider clan charge, or the objective will fail to generate properly for the ninja (see #32864)

TurboTrackerss14 commented on Resources/Prototypes/Entities/Markers/Spawners/mobs.yml:
```diff
@@ -883,7 +883,7 @@
 
 - type: entity
   name: xeno queen entity spawner
```
xeno praetorian entity spawner :trollface:

Kadeo64 commented:
can you cognizine the praetorian?

SlamBamActionman commented:
> can you cognizine the praetorian? that's my only question

Yesn't! There are some issues with the Praetorian being unrevivable so I gotta look into that but we did have a playtest round where the crew managed this (with minor admin intervention to make the revival work).

Kadeo64 commented:
> Yesn't! There are some issues with the Praetorian being unrevivable so I gotta look into that but we did have a playtest round where the crew managed this (with minor admin intervention to make the revival work).

does it not metabolize reagents? I know rmc14 xenos don't but i've only fought xenos like thrice in upstream

SlamBamActionman commented:
> does it not metabolize reagents? I know rmc14 xenos don't but i've only fought xenos like thrice in upstream

It was more a case of the defib not working - I just need to look into whatever mob setting I clicked wrong to make it possible to revive. But the idea is the cog xeno will be possible!

Kadeo64 commented:
Also, this might be worth looking into, but the 'Exomorph' name might be a little bit outdated considering the praetorian sprite used is most likely going to be recognized as a **_Xenonid_**, from RMC14.

IProduceWidgets commented:
![image](https://github.com/user-attachments/assets/23e97846-b829-4ce6-b3ec-820a789cb3ea)
Normally this is separated from the hull-wall so a single explosion doesn't cause uberpressure by breaking the wall and allowing the miners to send the room to 4500k might be fine though idk.

![image](https://github.com/user-attachments/assets/fe224c63-8e2e-4467-8b05-6fb44ca3caf6)
This feels very very smol. Also ideally TEG has a sort-of round start setup. Doesn't need to be perfect, but you don't want players spending 45 minutes laying pipe.

[Not pictured]
I would avoid lots of doubled walls.

![image](https://github.com/user-attachments/assets/7e72c672-f8ea-4bdf-9811-1217a804c743)
AME in the middle of the hallway is too easy to sabotage. You can just disassemble it and then it explodes.

![image](https://github.com/user-attachments/assets/9b3c376a-5303-41bf-a8ff-f0a8caa17f58)
defib cabinet on window

Needs red phone

![image](https://github.com/user-attachments/assets/90e5930a-b967-4ac5-bcfd-25565818b34b)
Use mail system instead

![image](https://github.com/user-attachments/assets/ff1c9bc9-624d-4942-85e0-cb5d3975787e)
These will have to be special containment field generators with their range edited. Also, I think its likely to sound really annoying.

![image](https://github.com/user-attachments/assets/c61444e6-97b1-466d-8460-abcd6a961d46)
![image](https://github.com/user-attachments/assets/b93e6467-92bd-469f-ac75-8b8738574a0f)
wallmounts on windows

IProduceWidgets commented:
I havent tried it in game yet to check all the things, I will soonish.

SlamBamActionman commented:
Thanks for checking it out!

> Normally this is separated from the hull-wall so a single explosion doesn't cause uberpressure by breaking the wall and allowing the miners to send the room to 4500k might be fine though idk.

With the space around the outside the only worry is if atmos get uberpressured, and atmos is the one department capable of handling it. Might be an issue, but hasn't come up with testing nor been a concern by any atmos mains using it.

> This feels very very smol. Also ideally TEG has a sort-of round start setup. Doesn't need to be perfect, but you don't want players spending 45 minutes laying pipe.

I'm not an atmos main, so I'm definitely up to hear suggestions on how to improve the roundstart atmos setup!

> [Not pictured] I would avoid lots of doubled walls.

The station and more importantly wall texture was designed to be able to support doubled walls. While I know this is a mapping faux pas on normal maps, this was done very intentionally and deliberately here, with adjustment in wall thickness to certain areas to account for it. Part of the design is that much of the station lacks good meteor protection, which is offset by the wall thickness. 

> AME in the middle of the hallway is too easy to sabotage. You can just disassemble it and then it explodes.

This is intentional and explained in the Engineering under the "Specific Area Descriptions" section.

>  defib cabinet on window

This might not be an issue soon, depending on whether #33165 gets merged.

>  Use mail system instead

The station was designed with the intention to allow traversal via the bins, should it be necessary. This is most common in the Medbay/Chemistry loop, but I want to keep the others as an option.

> These will have to be special containment field generators with their range edited. Also, I think its likely to sound really annoying.

They are extended range generators, and as described in the Perma section of "Specific Area Descriptions", it's been designed to ensure the noise can be avoided by being in the west part of Perma. The original design had much more of a noise issue and the complaint has not come up since the redesign.

> The nuke will be unable to be reanchored here due to space proximity.

This should not be a blocker since there are multiple maps that have vaults with space too close in proximity.

> Better to use airtight flaps and conveyors for this and the station so we can avoid the need of fans on station. Or if you _really_ want openable doors you could use docking ports which also solve the fan issue.

Fans are still being used in docking doors and just about any other commonly utilized space-facing airlock (e.g. arrivals, visitor and evac). I do not see this as an issue, especially since the goal is an easy-to-load shuttle stationside and conveyors/docking ports slow it down considerably.

> This will bring back the issue of throwing people into space on evac by preventing the doors from closing.

Prrrrobably fair. 😅 The doors aren't prevented from being closed, however the button to open them exists in the evac Bridge and shouldn't be too easy to activate by crew.

slarticodefast commented:
> I'm not an atmos main, so I'm definitely up to hear suggestions on how to improve the roundstart atmos setup!

The burn chamber could use an unlinked air alarm, which is usually needed for the scrubbers.
The atmos department seems ok to me in size, it is a little larger than that of Core for example. Fitting both the TEG setup and a frezon layout in there would need some planning. Depends a little on the player count you are going for, but it will be easier if we eventually get proper pipe layers.

Shaddap1 commented:
I like chem being moved.
Sec and sci really standout for great design.

As an engi main, why do you hate us hahaha.
I understand the impulse to create friction, but the AME supposed to be setup very quickly and easily, and only as a stopgap until PA/teg is running.  Requiring us to build a secure setup makes us a soft target for tiders, clowns, shitters, etc to antagonize us.  Sec is far away from this area, so the "friction" here is basically one or two engineers have to up that hall at roundstart and hope no one messes with them while they are building.  With the PA setup half done already, my instinct as a CE is to just say forget AME, everyone do tesla then go do solars.

Engineers don't want friction, they want to be efficient.  So **creating this AME vulnerability is creating a problem not worth solving**, and the AME isn't even a good target for legitimate antags in the first place.  The room east of PA with generators and emitters is very large and sparse for its purpose, so I suggest splitting it in half, and filling the second room with materials worth stealing.  That area already vulnerable to hacking in from maints and provides a direct path into the main room.

Shaddap1 commented:
Another note concerning med: the entrance is not very secure.  It seems way too easy to walk by and chuck a bomb into the main triage room from the hallway.  I think many med players want more security when it comes to patience admittance, as when you're on code red and the doors get emergency accessed, anyone can walk in and minibomb a body, steal from crit/dead people, etc.  Obviously there is some responsbility for a CMO to request security presence, but many maps have at least a one room buffer between hallway and triage and this does help with security.

Pixeltheaertist commented:
This looks really great, i'm quite excited to check it out when it gets merged! I'm a huge fan of those resin walls too... once this gets merged i gotta pull it downstream

SlamBamActionman commented:
![xeno-0](https://github.com/user-attachments/assets/6d305b30-96d3-4a93-b3da-428e0183b7d5)

Xeno has been updated to version 1.4

Changes include, but are not limited to:

**Mapping changes:**

- Changed Cargo's northwest area, making the flow into the actual Cargo area smoother for Cargo workers. Cargo has also been more incorporated visual-wise into the station to promote environmental storytelling.
- Reduced the number of 3-thick walls around the station.
- Slight adjustments to the Chapel area to increase readability.
- Bar maintenance has been changed and expanded with more interesting rooms.
- Singularity/Tesla generators moved indoors. Engineering slightly modified to account for dragging them outdoors.
- Cursed item is now more cursed.
- Minor medbay expansion.
- Minor RD room adjustment.

**Prototype changes:**

- New basic doors used for maintenance and public pathways. (Note: attribution will be included shortly)
- Gunarm shipguns have been replaced by a dedicated new prototype, the `Legacy PTK-150 "Matter Deconstructor"`.
  - Fire delay increased from 0.5s to 5s. 
  - Recharge time increased from 5s to 25s.
  - Max ammo lowered from 5 to 3.
  - Removed signal triggers Toggle, On and Off (making automatic firing impossible).
  - Increased density, causing slower drag speed.
  - No machine board; can't be rebuilt if destroyed. 
- Added a cosmetic broken cryopod, which can be scrapped for minor gain (20 steel, 10 glass). Rusted thrusters can now also be scrapped (12 steel).
- Added "dark squiggly tile" and "dark squiggly vertical tile". Can not be created (yet).
- Minor buffs to the maintenance Xeno to make it less of an absolute pushover.

SlamBamActionman commented:
### Looking for spriters/artists!

**The map has been given merge approval by Emisse**, on the condition that it receives custom art assets to better convey the "alien" aesthetic of the station. Problem is, I'm not a very good artist! So if you got the skills and want to see the map in the game, these are the art assets needed: 

- [ ] Xeno Steel Tile - A replacement for the dark steel tile; needs to work across the station without causing visual clutter. The current dark steel tile basically makes the station feel like a "normal station in dark mode" and this needs to change. It's very important that the tile isn't visually distracting. 
- [ ] Xeno Window - Should be somewhere between a normal and reinforced window defense-wise and replace the majority of plasma windows across the station. 
- [ ] Xeno Plating - A version of maints plating that can be put inside of maintenance. There is already a custom "dark techmaint", so probably something similar to that.
- [x] Xeno Organic Tile - A squishy-looking xeno texture. There is already "xeno flooring" which emulates alien tendril structures, but an alternative would be nice.
- [ ] Xeno Seethrough/Normal Door - Doors to use in hallways. These exist already, ported from RMC and edited, but they require touch-ups to be more visually fitting. 
- [ ] Xeno Blast Door - Blast doors to use in the station gunbays/disposals.
- [ ] Xeno Ship Turrets - To be used in the station gunbays. Can use the existing xeno turret as art reference.
- [ ] Xeno Table & Chair - Optional. Helps provide variety in station theming between areas.

Hit me up on Discord or post here/in the #cartography "xeno review thread" thread if you wanna help out! Map's almost there 👽 ❤️

SlamBamActionman commented:
![xeno-0](https://github.com/user-attachments/assets/fb362ccc-bc95-450c-89a9-5830ed41a973)

Xeno has been updated to version 1.5.
This version is NOT compatible with the current version of the game, as it relies on https://github.com/space-wizards/RobustToolbox/pull/5652 to support rotated tiles.

Changes include, but are not limited to:

Mapping changes:
- New xeno-themed steel and maints tiles.
- Changed around Science to be more thematically appropriate and have a nicer flow.
- Medical expanded eastward to provide more space.
- AI sat turrets.
- Cargo mail.
- Changed over to xeno-themed windows.
- Departmental economy computers.
- Slightly expanded the armory.
- Nuke containment widened.
- Xeno furniture.

Prototype changes:
- Xeno table, chair, window, blastdoor and shipgun added.
- Added new xeno tiles.

Current issues:
- Construction must be handled for all the new prototypes and objects.
- Resin window must have more reinforced window-esque properties.

SlamBamActionman commented:
Xeno Station is now in a merge-ready state!

While the map may be merged, please note that https://github.com/space-wizards/space-station-14/pull/37466 should ideally be merged before the map gets playtested, as the bug being addressed impacts certain parts of Xeno's playability quite heavily.

ArtisticRoomba commented:
the ultimate trojan horse of a map review
"hmm today i will review xeno station"
>look at diff
>176 files

ArtisticRoomba commented on Resources/Prototypes/Entities/Structures/Shuttles/cannons.yml:
```diff
@@ -360,3 +360,59 @@
+    availableModes:
+    - SemiAuto
+    soundGunshot:
+      path: /Audio/Weapons/Guns/Gunshots/kinetic_accel.ogg
```
use sound collections for captioning support in the future

ArtisticRoomba reviewed (changes requested):
prototype review

ArtisticRoomba reviewed (changes requested):
Map review:
- Power needs to be balanced according to the [Power Generation Rework](https://docs.spacestation14.com/en/space-station-14/departments/engineering/proposals/power-generation.html) document. Generally, 30% battery level at 5 minutes with AME on, 20% at 10 minutes with solars on, 15% at most when the generators are supposed to kick on at the 20 minute mark.
- Air Alarm sprite needs to be pulled down from the wall to match its normal counterpart.
- Spacing virology activates the firelocks unnecessarily in the other room connected to it.
- Mapped plant pot and discount dans inside of each other near arrivals.
- RD server room needs freezer and cooling system.
- Roboticist room generally needs more going for it. Try posters, a sink, morgue, latex gloves, surgical tools laid out on the table, etc.
- Do not map air vents and scrubbers underneath vending machines or tables in order to hide them, they should be intentionally visible. See QM's room.
- Try to avoid having separated engineering/atmospherics voltage networks as they can both screw each other over if only one side sets up power.
- Wastenet needs to be colored.
- Direct airlock to an active burn chamber is a super dangerous move. I would put a button to at least bolt it at least, completely add a second airlock system or make it only space-accessible at most.
- Air miner windows can burst and overpressure atmos easily.
- Chef freezer needs freezer atmospherics system
- Interrogation room needs signage
- Remove mapped mini jetpack outside
- AI SMES unit can be easily sniped by its own turrets and instantly trolls itself when it should be a secure area
- AI can be sniped with a laser weapon through its own glass windows, we don't want this to happen. Make them only visible from a non-windowed area, AKA add blast doors on the west face.
- Remove tiles under girder in cargo shuttle to expose hull plating, it looks ugly right now IMO
- Radiation collectors need to be moved inside, see bagel or box
- Bridge substation behind command access locked comms console is kinda lame, engineers should be able to mess with it or repair it without having to bust down a plasma window.
- Evac shuttle needs to be a fair amount bigger than what you've got right now, and it also needs to have a robust atmospherics system. See Oasis' evac for a good example.
- Redundant levers for controlling the brig imprisonment field seem to be mapped?
- Two-way-lever on disposals system that opens the outer door should close the door when the lever returns to the middle position instead of toggling the door
- Window to tech vault should exist so engies dont get trolled looking for it (they'll run away when they see the `Atmos` sign
- There should be oxygen/nitrogen canisters in EVA
- Departmental economy computers need to be mapped.

Other than that, such a good map.

SlamBamActionman commented:
> Map review:

All points have now been addressed (or at least attempted). Let me know if there's anything remaining. 

For power, due to the different power sources feeding into different areas, it wasn't entirely easy to ascertian if the map is following the power document. It seems to be correct (e.g. with no power, all SMESes get down to ~35%) and following the AME/solar steps at the provided times seems to prolong power decently into the run, although perhaps not to the desired percentages.

Since the map has a robust pre-setup for Singularity and Tesla I'd still like to playtest it to see how it feels in the hands of players.

ArtisticRoomba commented:
I'll give it a test but I'm sure it's A-OK. I'm planning on buffing Engi soon anyways.

ArtisticRoomba reviewed (approved):
![image](https://github.com/user-attachments/assets/c36f75d4-14a7-41e7-9811-7cfd1ea6416a)

ArtisticRoomba commented:
Let me know when https://github.com/space-wizards/space-station-14/pull/37466 is in and I'll merge this

ArtisticRoomba commented:
It's peak time.

kosticia commented:
DAMN🎉🎉🎉

Kiri-Yoshikage commented:
Wait, so if the power dies all the perma prisoners can just walk out?.. Security has some time to prepare, of course, as it's not an instant process, but still. Feels like most of the times sec will just ask for normal walls to be built because of that
